### ISSUE TITLE
Unlocked images in the gallery are not saved if you close the game incorrectly. (7.4.11)

### ISSUE THREAD

**Issue by maLikiz** [renpy/renpy#3278]
Unlocked images in the gallery are not saved if you close the game incorrectly.

OS: Windows/Android.
Renpy version: (7.4.11)

How to reproduce:
* Add your own images under the names specified in the code below.
* Delete persistent.
* Start the game, go to the second picture, you will see that both images have unlocked in the gallery.
* Open Task Manager in Windows (or in your OS) and end `pythonw.exe` task.
* Open the game again and you will see that the images in the gallery are no longer available.

Code example:
```rpy
image image_1 = 'image_1.jpg'
image image_2 = 'image_2.jpg'

init python:
    g = Gallery()

    g.button("image_1")
    g.unlock_image("image_1")

    g.button("image_2")
    g.unlock_image("image_2")

    g.locked_button = im.Scale("locked.jpg", 200, 200)

screen gallery:
    grid 1 2:
        xfill True
        yfill True
        ymaximum 420

        add g.make_button("image_1", im.Scale("image_1.jpg", 200, 200))
        add g.make_button("image_2", im.Scale("image_2.jpg", 200, 200))

label start:
    show screen gallery

    "Start label"
    "During the passage, do not restart the game and do not go to the main menu"
    "If you go to the menu, the gallery will be saved"

    show image_1
    "image_1"

    show image_2
    "image_2"

    "Open Task Manager in Windows (or in your OS) and end 'pythonw.exe' task"

    return
```

If I understand correctly, the persistent file is saved only if the game was closed correctly or if the player goes to the main menu. Is it possible to update a persistent file every time a player reaches an image to unlock it in the gallery? Will this be the right behavior or are there possible consequences?

**Comment by renpytom** [comment]
I think that would kill performance, so I wouldn't do that. We do save persistent each time a game is saved on Android, so it might make sense to do that on other platforms, as a compromise between not saving the game until quit and saving it everytime a new image, label, audio file, etc is shown. 

Fundamentally, though, ending a game using Task Manager is asking for trouble, when supported ways of exiting a game exist.

**Comment by maLikiz** [comment]
> We do save persistent each time a game is saved on Android

It doesn't seem to be working. I did a test in Bluestacks 5, I unlocked images in the gallery, then saved the game, and killed Bluestacks 5 in the task manager. Then I started the game again and saw that the previously unlocked images were now locked.

Why I created this issue: sometimes the game crashes for players, I can't say exactly what the problem is, maybe the problem is memory leaks. Also, players reported that the gallery is not saved. Therefore, in my opinion, when the game closes incorrectly, it forgets which images were unlocked.
